src/Endpoint/Partition.php

```diff
@@ -161,7 +161,8 @@ private function resolveRegion($service, $region)
     private function isServicePartitionGlobal($service)
     {
         return isset($this->data['services'][$service]['isRegionalized'])
-            && false === $this->data['services'][$service]['isRegionalized'];
+            && false === $this->data['services'][$service]['isRegionalized']
+            && isset($this->data['services'][$service]['partitionEndpoint']);
```

Comment:
Looks good, just curious, is this fix a work-around for some unexpected global endpoints or this is a normal behavior?